ssl/s3_lib.c

```diff
@@ -3564,16 +3565,11 @@ long ssl3_ctrl(SSL *s, int cmd, long larg, void *parg)
         break;
 
     case SSL_CTRL_GET_TLSEXT_STATUS_REQ_OCSP_RESP:
-        *(unsigned char **)parg = sc->ext.ocsp.resp;
-        if (sc->ext.ocsp.resp_len == 0
-                || sc->ext.ocsp.resp_len > LONG_MAX)
-            return -1;
-        return (long)sc->ext.ocsp.resp_len;
+        *(STACK_OF(OCSP_RESPONSE) **)parg = sc->ext.ocsp.resp;
```

Comment:
You can't change the code behind a SSL_CTRL like this.
It will break any existing code calling it.

This probably breaks some test code.